### PR TITLE
feat(webpaas): add tracking to web paas

### PR DESCRIPTION
**Pull request by varun257**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/web-paas-prod`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-5671
| License          | BSD 3-Clause

## Description

add navigation and operation tracking to webpaas

### CODE REVIEW

File: packages/manager/modules/web-paas/src/components/delete-user/controller.js [hunk 1]
```diff
@@ -2,16 +2,31 @@ import get from 'lodash/get';
...
   }
 
   $onInit() {
     this.isLoading = false;
   }
 
+  cancel() {
+    this.atInternet.trackClick({
+      name:
+        'web-paas::project-user-license::user-table-options::delete-user-cancel',
```
Comment (radireddy) [inline packages/manager/modules/web-paas/src/components/delete-user/controller.js:18]: in some places tracking name is starting with "web-paas-platform-sh". Here it starts with 'web-paas'. Is it ok?

File: packages/manager/modules/web-paas/src/components/delete-user/controller.js [hunk 2]
```diff
@@ -2,16 +2,31 @@ import get from 'lodash/get';
...
+      type: 'action',
+    });
+    return this.goBack();
+  }
+
   delete() {
+    this.atInternet.trackClick({
+      name:
+        'web-paas::project-user-license::user-table-options::delete-user-confirm',
```
Comment (radireddy) [inline packages/manager/modules/web-paas/src/components/delete-user/controller.js:27]: Can you extract common prefix and reuse it in all places. Refer kubernetes for code
/manager/modules/pci/src/projects/project/kubernetes/kubernetes.routing.js

kubeTrackPrefix: () => 'PublicCloud::pci::projects::project::kubernetes',

      sendKubeTrack: /* @ngInject */ (kubeTrackPrefix, trackClick) => (
        complement,
      ) => {
        trackClick(`${kubeTrackPrefix}::${complement}`);
      },

      trackClick: /* @ngInject */ (atInternet) => (hit) => {
        atInternet.trackClick({
          name: hit,
          type: 'action',
        });
      },

File: packages/manager/modules/web-paas/src/details/service/add-addon/controller.js [hunk 3]
```diff
@@ -127,13 +132,21 @@ export default class {
...
+      case ADDON_TYPE.ENVIRONMENT:
         this.presentCount = this.project.getTotalEnvironment() - 2;
         this.description = this.$translate.instant(
           'web_paas_service_add_staging_description',
           { environment: this.presentCount },
         );
+        this.trackCancelText =
+          'web-paas::project-dashboard::setup-additional-environments::cancel';
+        this.trackConfirmText =
+          'web-paas::project-dashboard::setup-additional-environments::confirm-';
```
Comment (radireddy) [inline packages/manager/modules/web-paas/src/details/service/add-addon/controller.js:149]:
```suggestion
          'web-paas::project-dashboard::setup-additional-environments::confirm';
```

File: packages/manager/modules/web-paas/src/details/service/add-addon/controller.js [hunk 4]
```diff
@@ -153,10 +166,22 @@ export default class {
         )} ${this.$translate.instant(
           'web_paas_service_add_license_description2',
         )}`;
+        this.trackCancelText =
+          'web-paas::project-dashboard::setup-additional-licences::cancel';
+        this.trackConfirmText =
+          'web-paas::project-dashboard::upgrade-licences::confirm-';
```
Comment (radireddy) [inline packages/manager/modules/web-paas/src/details/service/add-addon/controller.js:172]:
```suggestion
          'web-paas::project-dashboard::upgrade-licences::confirm';
```

File: packages/manager/modules/web-paas/src/details/service/service.controller.js [hunk 5]
```diff
@@ -6,6 +6,18 @@ export default class WebPassDetailsServiceCtrl {
...
+    this.WEB_PAAS_TRACK_CHANGE_OFFER =
+      'web-paas::project-dashboard::change-range';
+    this.WEB_PAAS_TRACK_CHANGE_VCPU =
+      'web-paas::project-dashboard::setup-vcpus';
+    this.WEB_PAAS_TRACK_ADDON_STORAGE =
+      'web-paas::project-dashboard::upgrade-storage';
+    this.WEB_PAAS_TRACK_ADDON_LICENSE =
+      'web-paas::project-dashboard::setup-user';
+    this.WEB_PAAS_TRACK_ADDON_ENVIRONMENT =
+      'web-paas::project-dashboard::setup-additional-environments';
```
Comment (radireddy) [inline packages/manager/modules/web-paas/src/details/service/service.controller.js:20]: Here, "web-paas::project-dashboard::" is a common prefix which can be extracted to route the way its done in kubernetes.

**Review by antleblanc — CHANGES_REQUESTED** [review]
Hi @varun257!

In addition to @radireddy comments, could you please amend your commit and prefer using a signed commit?

Thank you!

File: packages/manager/modules/web-paas/src/components/delete-user/controller.js (hunk 1 above)
Comment (varun257) [reply packages/manager/modules/web-paas/src/components/delete-user/controller.js:18]: The ones which are directing to platform sh links have  "web-paas-platform-sh" as tracking so its okay

**Comment by marie-j** [comment]
Hello @varun257 , 

Could you please solve existing conflicts ?

Thanks

File: packages/manager/modules/web-paas/src/web-paas.service.js [hunk 6]
```diff
@@ -189,6 +180,18 @@ export default class WebPaasService {
...
+      accountName,
+    });
+  }
+
+  deleteUser(projectId, customerId) {
+    return this.$http.delete(
+      `/webPaaS/subscription/${projectId}/customer/${customerId}`,
+    );
+  }
+
```
Comment (JeremyDec) [inline packages/manager/modules/web-paas/src/web-paas.service.js:194]: Hello @varun257, I see this logic as part of the commit "[...] add tracking". Is this related to a feature ? Should it be done in another PR/commit ?

File: packages/manager/modules/web-paas/src/add/add.controller.js [hunk 7]
```diff
@@ -246,4 +265,12 @@ export default class WebPassAddCtrl {
...
       });
   }
+
+  isEditable() {
+    return !this.isChangingPlan();
+  }
+
+  isChangingPlan() {
+    return this.selectedProject;
+  }
```
Comment (JeremyDec) [inline packages/manager/modules/web-paas/src/add/add.controller.js:275]: Same here, this code seems to not be part of the tracking request.

File: packages/manager/modules/web-paas/src/web-paas.service.js (hunk 6 above)
Comment (varun257) [reply packages/manager/modules/web-paas/src/web-paas.service.js:194]: This also included fixes for some qa issues @JeremyDec, i have added it in description of the commit

File: packages/manager/modules/billing/src/components/subscription-tile/subscription-tile.controller.js [hunk 8]
```diff
@@ -23,16 +23,13 @@ export default class ServicesActionsCtrl {
   }
 
   $onInit() {
-    this.disableServiceActions = this.$attrs.disableServiceActions
-      ? this.disableServiceActions
-      : false;
+    this.displayServiceActions = this.coreConfig.isRegion(['EU', 'CA']);
```
Comment (marie-j) [inline packages/manager/modules/billing/src/components/subscription-tile/subscription-tile.controller.js:26]: Looks strange that this appears in the `diff`, it doesn't look related to webpaas

File: packages/manager/modules/web-paas/src/details/user-licences/user-licences.controller.js [hunk 9]
```diff
@@ -22,4 +31,31 @@ export default class {
...
+    set(
+      this.CHANGE_OFFER_IMPRESSION_TRACKING_DATA,
+      'variant',
+      `[${this.project.offerName}]`,
+    );
+    set(
+      this.CHANGE_OFFER_IMPRESSION_TRACKING_DATA,
+      'format',
+      `[${this.project.isStartOffer() ? 'develop-expand' : 'expand'}]`,
+    );
```
Comment (marie-j) [inline packages/manager/modules/web-paas/src/details/user-licences/user-licences.controller.js:45]: Is it needed to override some values, could we extend instead 
```suggestion
    const trackingData = {
    ...CHANGE_OFFER_IMPRESSION_TRACKING_DATA,
    variant: ...
    }
```

File: packages/manager/modules/web-paas/src/details/user-licences/user-licences.controller.js [hunk 10]
```diff
@@ -1,11 +1,20 @@
...
 
 export default class {
+  /* @ngInject */
+  constructor(atInternet) {
+    this.atInternet = atInternet;
+  }
+
   $onInit() {
     this.ADDON_TYPE = ADDON_TYPE;
+    this.CHANGE_OFFER_IMPRESSION_TRACKING_DATA = CHANGE_OFFER_IMPRESSION_TRACKING_DATA;
```
Comment (marie-j) [inline packages/manager/modules/web-paas/src/details/user-licences/user-licences.controller.js:13]: Do we need to add it as a property, it seems only used in controller so we can directly use imported value

File: packages/manager/modules/web-paas/src/web-paas.service.js [hunk 11]
```diff
@@ -504,4 +506,11 @@ export default class WebPaasService {
       },
     );
   }
+
+  trackClick(name) {
+    this.atInternet.trackClick({
+      name,
+      type: 'action',
+    });
+  }
```
Comment (marie-j) [inline packages/manager/modules/web-paas/src/web-paas.service.js:515]: Is it really needed ?

**Review by radireddy — APPROVED** [review]

**Review by ganeshkumar1989 — APPROVED** [review]

**Review by marie-j — APPROVED** [review]
